src/less/star-rating/base/star-rating.less

```diff
@@ -10,3 +15,33 @@ svg.star-rating {
     vertical-align: middle;
     width: 64px;
 }
+
+div.star-rating svg.star_rating_icon {
+    fill: none;
+    height: 12px;
+    width: 12px;
+}
+
+/* each star starts unfilled */
+div.star-rating svg.star_rating_icon use.star_rating_icon_symbol {
+    fill: @star-unfilled;
+}
+
+/* full stars */
+div.star-rating[data-stars^="1"] svg:nth-child(-n + 1) use.star_rating_icon_symbol,
+div.star-rating[data-stars^="2"] svg:nth-child(-n + 2) use.star_rating_icon_symbol,
+div.star-rating[data-stars^="3"] svg:nth-child(-n + 3) use.star_rating_icon_symbol,
+div.star-rating[data-stars^="4"] svg:nth-child(-n + 4) use.star_rating_icon_symbol,
+div.star-rating[data-stars^="5"] svg:nth-child(-n + 5) use.star_rating_icon_symbol {
+    fill: @star-filled;
+}
+
+/* half stars */
+div.star-rating[data-stars="0-5"] svg:nth-child(1) use.star_rating_icon_star_half,
+div.star-rating[data-stars="1-5"] svg:nth-child(2) use.star_rating_icon_star_half,
+div.star-rating[data-stars="2-5"] svg:nth-child(3) use.star_rating_icon_star_half,
+div.star-rating[data-stars="3-5"] svg:nth-child(4) use.star_rating_icon_star_half,
+div.star-rating[data-stars="4-5"] svg:nth-child(5) use.star_rating_icon_star_half {
+    .svgHalfXClip();
```

Comment:
I see the `svgHalfXClip` mixin is only being used once. So maybe just remove the mixin and enter the style directly here. Unless you foresee it being useful in other places. Generally though we try and steer clear of creating new public mixins (in fact we are leaning towards removing the last few we have left)